File: python/exercises/code-smells/feature-envy/src/product.py
from dataclasses import dataclass


@dataclass
class Product:
    id: str
    name: str
    price: float
    category: str
    weight: float
    fragile: bool = False
    manufacturer: str = ""

    def get_id(self) -> str:
        return self.id

    def get_name(self) -> str:
        return self.name

    def get_price(self) -> float:
        return self.price

    def get_category(self) -> str:
        return self.category

    def get_weight(self) -> float:
        return self.weight

    def is_fragile(self) -> bool:
        return self.fragile

    def get_manufacturer(self) -> str:
        return self.manufacturer

File: typescript/exercises/code-smells/feature-envy/src/Product.ts
export class Product {
  constructor(
    private readonly id: string,
    private readonly name: string,
    private readonly price: number,
    private readonly category: string,
    private readonly weight: number,
    private readonly fragile: boolean = false,
    private readonly manufacturer: string = ''
  ) {}

  getId(): string {
    return this.id;
  }

  getName(): string {
    return this.name;
  }

  getPrice(): number {
    return this.price;
  }

  getCategory(): string {
    return this.category;
  }

  getWeight(): number {
    return this.weight;
  }

  isFragile(): boolean {
    return this.fragile;
  }

  getManufacturer(): string {
    return this.manufacturer;
  }
}
